sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Watch.java

```diff
@@ -898,27 +909,31 @@ public String toString(Growth.TerminationCondition<?, TerminationStateT> termina
 
     @Override
     public synchronized GrowthState<OutputT, KeyT, TerminationStateT> checkpoint() {
+      checkState(
```

Comment:
<!--new_thread; commit:b04f5ed31c0be1f231dd7c1613ea50d31616b127; resolved:0-->
Should we be rejecting the checkpoint request instead of failing here ?

Reply:
Rejecting the checkpoint is not allowed. Allowing it is one of the alternatives I considered, but since runner needs access to positions anyway, I preferred to just do that.